### ISSUE TITLE
RFC: IPC Store

### ISSUE THREAD

**Issue by CanadaHonk** [gluon-framework/gluon#32]
Proposal for adding a "Store" to IPC, accessible to both Node and Web, for easily sharing common data between them. Not persistent. Example usage:

## Proposal 1 - Async getter/getter
Web would fetch config key from IPC when gotten, hence being async.

```js
// Node backend
Window.ipc.store.config = {
  env: 'prod'
};
```

```js
// Web frontend
const runningEnv = (await Gluon.ipc.store.config).env;
```

## Proposal 2 - Sync getter/getter
Web would be sent updated Store key and values whenever they are set, hence being sync.

```js
// Node backend
Window.ipc.store.config = {
  env: 'prod'
};
```

```js
// Web frontend
const runningEnv = Gluon.ipc.store.config.env;
```

## Proposal 3 - Async Functions
```js
// Node backend
Window.ipc.store.set('config', {
  env: 'prod'
});
```

```js
// Web frontend
const runningEnv = (await Gluon.ipc.store.get('config')).env;
```

## Proposal 4 - Sync Functions
```js
// Node backend
Window.ipc.store.set('config', {
  env: 'prod'
});
```

```js
// Web frontend
const runningEnv = Gluon.ipc.store.get('config').env;
```

Please comment with which proposal(s), you prefer! Open to implementing 1/2 and 3/4.

**Comment by Beefers** [comment]
Same here, 2 is cleanest imo.

**Comment by CamjamPNG** [comment]
for anyone who doesn't
 understand the code here's what it is in English

![image_2023-01-07_114535068](https://user-images.githubusercontent.com/38945005/211161392-596d1394-a8d4-41bb-a8d0-6557e14894b4.png)

**Comment by CanadaHonk** [comment]
Proposals 2 and 4 were implemented in v0.11.0
